Userland/Libraries/LibJS/Runtime/ECMAScriptFunctionObject.h

```diff
@@ -152,6 +152,10 @@ class ECMAScriptFunctionObject final : public FunctionObject {
     bool m_arguments_object_needed { false };
     Vector<VariableNameToInitialize> m_var_names_to_initialize_binding;
     Vector<DeprecatedFlyString> m_function_names_to_initialize_binding;
+
+    size_t m_function_environment_bindings_count { 0 };
+    Optional<size_t> m_var_environment_bindings_count;
+    Optional<size_t> m_lex_environment_bindings_count;
```

Comment:
Since these should always have a value, I don't think they need to be `Optional<size_t>`. Just `size_t` is fine.

Reply:
I made them optional because those are sizes for environments are not necessary allocated during `function_declaration_instantiation`. This is confusing indeed so I changed it to `size_t`.